flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/mysql/MySQLDialect.java

```diff
@@ -29,7 +30,7 @@
 import java.util.stream.Collectors;
 
 /** JDBC dialect for MySQL. */
-public class MySQLDialect extends AbstractDialect {
+class MySQLDialect extends AbstractDialect {
```

Comment:
nit: `@Internal` to be clear?

Reply:
that's fine. I don't think we usually annotate things that are package-private, as it's kind of implied.

Reply:
I think rather we don't have strong enough rules on this in the project, see https://github.com/apache/flink/pull/17186#discussion_r705312290.

For my architectural tests I'll have to address this with the community somehow, too.